tcks/apis/messaging/messaging-inside-container/src/main/java/com/sun/ts/tests/jms/ee20/cditests/usecases/BMBean1IFProxy.java

```diff
@@ -0,0 +1,80 @@
+package com.sun.ts.tests.jms.ee20.cditests.usecases;
+
+import com.sun.ts.lib.harness.RemoteStatus;
+import com.sun.ts.lib.util.TestUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.StringWriter;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Properties;
+
+public class BMBean1IFProxy implements InvocationHandler {
+
+    private Properties testProps;
+
+    public static BMBean1IF newInstance(Properties testProps) {
+        return (BMBean1IF) Proxy.newProxyInstance(
+            BMBean1IF.class.getClassLoader(),
+                new Class[] { BMBean1IF.class },
+                new BMBean1IFProxy(testProps));
+    }
+
+    private  BMBean1IFProxy(Properties testProps) {
+        this.testProps = testProps;
+    }
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+        String methodName = method.getName();
+
+        boolean ok = false;
+        if(methodName.equals("init")) {
+            //sendPost();
+            ok = true;
+        } else if(methodName.startsWith("cleanupQueue")) {
+            // RemoteStatus status = sendGet("removeTestData");
+            ok = true;
+        } else if(methodName.equals("toString")) {
+            return Client.class.getName()+"-proxy";
+        } else if(methodName.startsWith("method")) {
+            // RemoteStatus status = sendGet(methodName);
```

Comment:
@starksm64 Can you please suggest how can I invoke the Bean method here as required.

Reply:
Have your proxy class extend the com.sun.ts.tests.common.vehicle.none.proxy.AbstractBaseProxy class from jakarta.tck:common and implement the runTests(String) method. See this example:

tcks/apis/persistence/persistence-inside-container/platform-tests/src/main/java/ee/jakarta/tck/persistence/ee/packaging/ejb/descriptor/IClientProxy.java

Reply:
You will then need to update test client to include the appclient proxy war like is done in tcks/apis/persistence/persistence-inside-container/platform-tests/src/main/java/ee/jakarta/tck/persistence/ee/packaging/ejb/descriptor/ClientTest.java:

```java
            // non-vehicle appclientproxy invoker war
            WebArchive appclientproxy = ShrinkWrap.create(WebArchive.class, "appclientproxy.war");
            appclientproxy.addClasses(Client.class, ClientServletTarget.class, ServletNoVehicle.class);
            appclientproxy.addAsWebInfResource(new StringAsset(""), "beans.xml");
```

Reply:
I made commit to use IClientProxy similar to persistence packaging/ejb/descriptor. The changes still dont work for me. I get errors like below
> com.sun.ts.tests.common.vehicle.none.proxy.ServletDispatcher sendGet
> SEVERE: Failed to send beanUseCaseJ(), status code 404, content length: Optional.empty
> FAILED: beanUseCaseJ
> beanUseCaseJ failed
> , additional stuats:
> RemoteStatus: Failed. Failed to read response body Error: 'invalid stream header: 3C21444F'

@starksm64 Can you please check the jms/ee20/cditests/usecases/ folder changes.